parser/src/lexer.rs

```diff
@@ -29,22 +44,25 @@ impl<'input> Lexer<'input> {
             indentation_stack: vec![0],
             pending: Vec::new(),
             chr0: None,
-            location: 0,
+            location: Location::new(0, 0),
             chr1: None,
         };
         lxr.next_char();
         lxr.next_char();
+        // Start at top row (=1) left column (=1)
+        lxr.location.row = 1;
```

Comment:
What's the thinking behind 1-indexing rather than 0-indexing?

Reply:
row indicates the sourcecode row. Sourcecode starts at row 1.

Reply:
Rust consider files and lines 0-indexed though, so I worry that we'll end up subtracting 1 from columns and rows everywhere other than the code which actually presents the line and column numbers to users. (Or, rather, I'm worried that we _won't_ and that will lead to bugs.)

I would prefer to do 0-indexing internally, and then add 1 when we're finally displaying something to users.

(That said, I'm happy to stick with this; we can always revisit it in the future if necessary.)